assets/js/components/wp-dashboard/WPDashboardPopularPagesGA4.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * WPDashboardPopularPagesGA4 component.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { cloneDeep } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import {
+	MODULES_ANALYTICS_4,
+	DATE_RANGE_OFFSET,
+} from '../../modules/analytics-4/datastore/constants';
+import { ZeroDataMessage } from '../../modules/analytics/components/common';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import PreviewTable from '../PreviewTable';
+import TableOverflowContainer from '../TableOverflowContainer';
+import ReportTable from '../ReportTable';
+import DetailsPermaLinks from '../DetailsPermaLinks';
+import { numFmt } from '../../util';
+const { useSelect, useInViewSelect } = Data;
+
+export default function WPDashboardPopularPagesGA4( props ) {
+	const { WidgetReportError } = props;
+
+	const isGatheringData = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).isGatheringData()
+	);
+
+	const reportArgs = useSelect(
+		WPDashboardPopularPagesGA4.selectReportArgs,
+		[]
+	);
+
+	const report = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( reportArgs )
+	);
+
+	const titles = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getPageTitles( report, reportArgs )
+	);
+
+	const error = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getErrorForSelector( 'getReport', [
+			reportArgs,
+		] )
+	);
+
+	const loading = useSelect( ( select ) => {
+		const reportLoaded = select(
+			MODULES_ANALYTICS_4
+		).hasFinishedResolution( 'getReport', [ reportArgs ] );
+
+		const hasLoadedPageTitles = undefined !== error || undefined !== titles;
+
+		return ! hasLoadedPageTitles || ! reportLoaded;
```

Comment:
Obviously if page titles haven't been loaded yet, then we can return `true` early and not call the selector, right?